### PR TITLE
Add swippy swappys. typescript

### PR DESCRIPTION
**Pull request by Craiting**

### CODE REVIEW

File: lib/apps.js [hunk 1]
```diff
@@ -3,84 +3,91 @@
...
-                    if (err) {
-                        die(err);
-                    }
-
-                    _trackBot(bot);
-
-                    if (onInstallation) onInstallation(bot, config.createdBy);
-                });
-            }
+  configure: function(port, clientId, clientSecret, config, onInstallation) {
```
Comment (mahowa) [inline lib/apps.js]: NIT but we should convert this whole file to arrow syntax ()=>{}

File: package.json [hunk 2]
```diff
@@ -25,10 +26,14 @@
...
+    "@types/node": "^13.7.1",
     "axios": "^0.19.0",
     "botkit": "0.2.2",
+    "botkit-storage-mongo": "^1.0.7",
     "dotenv": "^8.2.0",
     "node-cron": "^2.0.3",
-    "redis": "^2.8.0"
+    "nvm": "0.0.4",
+    "redis": "^2.8.0",
+    "typescript": "^3.7.5"
```
Comment (mahowa) [inline package.json]: @types/node 
and
typescript 
can both be moved to devDependencies

File: src/helpers/ResyncPlayerStats.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+function ResyncPlayerStats(playerStats: any) {
+  axios({
+    method: "get",
+    url: `https://us-central1-tablechamp-444aa.cloudfunctions.net/stats`
+  })
+    .then((response: any) => {
+      playerStats
+      response.data.forEach((userStats: any) => {
+        // console.log('hi', userStats.stats);
```
Comment (mahowa) [inline src/helpers/ResyncPlayerStats.ts]: probably dont want this comment

**Comment by jnielson94** [comment]
@Craiting - What's the purpose in including the `build/` files in the repo? Are they needed for deploying?

File: src/helpers/GameHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    const p1Position = gameObj.black.p1.currPosition;
+    gameObj.black.p1.currPosition = gameObj.black.p2.currPosition;
+    gameObj.black.p2.currPosition = p1Position;
+  }
+}
+
+export async function initializeGame(mode: GameMode, playerList: Array<string>, revenge: boolean): Promise<GameObjType> {
+  if (!revenge) {
+    const formattedPlayers = playerList.join(',');
+    const getPositionsUrl = `https://us-central1-tablechamp-444aa.cloudfunctions.net/getPositions?users=${formattedPlayers}`;
```
Comment (jnielson94) [inline src/helpers/GameHelpers.ts]: This URL is different than one used in the finished game file, not sure how important that is though?

File: src/helpers/ActionHandler.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    p2: PlayerObj
+  },
+  black: {
+    score: number,
+    p1: PlayerObj,
+    p2: PlayerObj,
+  }
+}
+
+// TODO: Come back to this and finish it.
```
[the remark this answers is not in the document]
Comment (Craiting) [reply src/helpers/ActionHandler.ts]: No, this is saved for later.

File: package.json [hunk 6]
```diff
@@ -27,8 +28,14 @@
   "dependencies": {
     "axios": "^0.19.0",
     "botkit": "0.2.2",
+    "botkit-storage-mongo": "^1.0.7",
     "dotenv": "^8.2.0",
     "node-cron": "^2.0.3",
+    "nvm": "0.0.4",
```
[the remark this answers is not in the document]
Comment (Craiting) [reply package.json]: oops

File: src/FinishedGameMessage.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        p3: gameObj.mode === 'normal' ? "G" : "S",
+        p4: gameObj.mode === 'normal' ? "F" : "S",
+      }
+    }
+  };
+
+  axios({
+    method: 'post',
+    url: 'https://us-central1-gamerbot-beta.cloudfunctions.net/addScore',
+    // url: 'https://us-central1-tablechamp-444aa.cloudfunctions.net/addScore',
```
[the remark this answers is not in the document]
Comment (Craiting) [reply src/FinishedGameMessage.ts]: I'm gonna come back to this and make env variable swap between the two

File: src/FinishedGameMessage.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    data: payload,
+  }).then((response: any) => {
+    playerStats;
```
[the remark this answers is not in the document]
Comment (Craiting) [reply src/FinishedGameMessage.ts]: this was a weird thing I thought I had to do. I'll look into this again soon.

**Pull request merged**